File: solution/2149. Rearrange Array Elements by Sign/2149.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        queue<int> P, N;
        for (int x : nums) {
            if (x > 0)
                P.push(x);
            else
                N.push(x);
        }
        vector<int> res;
        for (int i = 0; i < nums.size(); ++i) {
            if (i % 2) {
                res.push_back(N.front());
                N.pop();
            } else {
                res.push_back(P.front());
                P.pop();
            }
        }
        return res;
    }
};

File: solution/2149. Rearrange Array Elements by Sign/2149.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums)
        i, j = 0, 1
        for x in nums:
            if x > 0:
                res[i] = x
                i += 2
            else:
                res[j] = x
                j += 2
        return res
    
